lib/monaco-adapter.js

```diff
@@ -1,301 +1,498 @@
-      if (op.isRetain()) {
-        index += op.chars;
-      } else if (op.isInsert()) {
-        var pos = this.monacoModel.getPositionAt(index);
-        range = new monaco.Range(
-          pos.lineNumber,
-          pos.column,
-          pos.lineNumber,
-          pos.column
-        );
-        id = { major: 1, minor: 1 };
-        ops = {
-          identifier: id, range: range, text: op.text,
-          forceMoveMarkers: true
-        };
-        this.monaco.executeEdits("my-source", [ops]);
-        index += op.text.length;
-      } else if (op.isDelete()) {
-        from = this.monacoModel.getPositionAt(index);
-        to = this.monacoModel.getPositionAt(index + op.chars);
+    /** Get co-ordinate position in Editor */
+    var start = this.monacoModel.getPositionAt(position);
+    var end = this.monacoModel.getPositionAt(selectionEnd);
 
-        range = new monaco.Range(from.lineNumber, from.column, to.lineNumber, to.column);
-        id = { major: 1, minor: 2 };
-        ops = {
-          identifier: id, range: range, text: "",
-          forceMoveMarkers: true
-        };
-        this.monaco.executeEdits("my-source", [ops]);
-      }
+    /** Selection is inversed */
+    if (position > selectionEnd) {
+      var _ref = [end, start];
+      start = _ref[0];
+      end = _ref[1];
     }
-    this.grabDocumentState();
-  };
 
-  MonacoAdapter.prototype.getValue = function () {
-    return this.monacoModel.getValue();
-  };
-
-  MonacoAdapter.prototype.getCursor = function () {
-    var e, e2, end, start, _ref, _ref1;
-    try {
-      selection = this.monaco.getSelection();
+    /** Add decoration to the Editor */
+    otherCursor.decoration = this.monaco.deltaDecorations(otherCursor.decoration,
+      [
+        {
+          range: new monaco.Range(
+            start.lineNumber, start.column,
+            end.lineNumber, end.column
+          ),
+          options: {
+            className: clazz
+          }
+        }
+      ]
+    );
 
-      if (typeof selection === 'undefined' || selection === null) {
-        selection = this.lastCursorRange;
+    /** Clear cursor method */
+    var _this = this;
+    return {
+      clear: function clear() {
+        otherCursor.decoration = _this.monaco.deltaDecorations(otherCursor.decoration, []);
       }
+    };
+  };
 
-      start = this.monacoModel.getOffsetAt(selection.getStartPosition());
-      end = this.monacoModel.getOffsetAt(selection.getEndPosition());
-    } catch (_error) {
-      e2 = _error;
-      console.log("Couldn't figure out the cursor range:",
-        e2, "-- setting it to 0:0.");
-      _ref = [0, 0], start = _ref[0], end = _ref[1];
-    }
+  /**
+   * @method registerCallbacks - Assign callback functions to internal property
+   * @param {function[]} callbacks - Set of callback functions
+   */
+  MonacoAdapter.prototype.registerCallbacks = function registerCallbacks(callbacks) {
+    this.callbacks = Object.assign({}, this.callbacks, callbacks);
```

Comment:
Can you explain why the move to `Object.assign` from straight assignment?

Reply:
So that if new param doesn't have action for some event, we can use the existing ones.